### PR TITLE
[MC-1279] CrowdStrike Falcon Intelligence plugin with 7 actions

### PR DESCRIPTION
**Pull request by mgrandys-r7**
### Description

Describe the proposed changes:

  - Initial plugin | Check Analysis Status, Download Artifact, Get Full Report, Get Reports IDs, Get Short Report, Get Submissions IDs, Submit Analysis

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Run

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/CrowdStrike Falcon https://example.com Step name: checkAnalysisStatus\nChecking analysis status of IDs: ['9de5069c5afe602b2ea0a04b66beb2c0_9de5069c5afe602b2ea0a04b66beb2c0', '9de5069c5afe602b2ea0a04b66beb2c0_9de5069c5afe602b2ea0a04b66beb2c0', '9de5069c5afe602b2ea0a04b66beb2c0_9de5069c5afe602b2ea0a04b66beb2c0', '9de5069c5afe602b2ea0a04b66beb2c0_9de5069c5afe602b2ea0a04b66beb2c0', '9de5069c5afe602b2ea0a04b66beb2c0_9de5069c5afe602b2ea0a04b66beb2c0']...\nGetting authentication token...\n",
    "meta": {},
    "output": {
      "submissions": [
        {
          "cid": "9de5069c5afe602b2ea0a04b66beb2c0",
          "created_timestamp": "2022-08-30T10:06:44Z",
          "id": "9de5069c5afe602b2ea0a04b66beb2c0_9de5069c5afe602b2ea0a04b66beb2c0",
          "origin": "uiproxy",
          "sandbox": [
            {
              "environment_id": 160,
              "sha256": "275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f",
              "submit_name": "https://example.com"
            }
          ],
          "state": "error",
          "user_id": "9de5069c5afe602b2ea0a04b66beb2c0",
          "user_name": "https://example.com",
          "user_uuid": "a149764d-21ee-42af-ac9f-023f5d23ab81"
        },
        {
          "cid": "9de5069c5afe602b2ea0a04b66beb2c0",
          "created_timestamp": "2022-08-30T09:25:35Z",
          "id": "9de5069c5afe602b2ea0a04b66beb2c0_9de5069c5afe602b2ea0a04b66beb2c0",
          "origin": "uiproxy",
          "sandbox": [
            {
              "environment_id": 300,
              "sha256": "275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f",
              "submit_name": "https://example.com"
            }
          ],
          "state": "error",
          "user_id": "9de5069c5afe602b2ea0a04b66beb2c0",
          "user_name": "https://example.com",
          "user_uuid": "a149764d-21ee-42af-ac9f-023f5d23ab81"
        },
        {
          "cid": "9de5069c5afe602b2ea0a04b66beb2c0",
          "created_timestamp": "2022-08-30T09:21:00Z",
          "id": "9de5069c5afe602b2ea0a04b66beb2c0_9de5069c5afe602b2ea0a04b66beb2c0",
          "origin": "uiproxy",
          "sandbox": [
            {
              "environment_id": 100,
              "sha256": "275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f",
              "submit_name": "https://example.com"
            }
          ],
          "state": "error",
          "user_id": "9de5069c5afe602b2ea0a04b66beb2c0",
          "user_name": "https://example.com",
          "user_uuid": "a149764d-21ee-42af-ac9f-023f5d23ab81"
        },
        {
          "cid": "9de5069c5afe602b2ea0a04b66beb2c0",
          "created_timestamp": "2022-11-04T13:56:17Z",
          "id": "9de5069c5afe602b2ea0a04b66beb2c0_9de5069c5afe602b2ea0a04b66beb2c0",
          "origin": "apigateway",
          "sandbox": [
            {
              "action_script": "default_randomfiles",
              "environment_id": 100,
              "network_settings": "offline",
              "submit_name": "my_submit_name",
              "system_date": "2022-09-22",
              "system_time": "12:55",
              "url": "hxxps://https://example.com"
            }
          ],
          "state": "error",
          "user_tags": [
            "example"
          ]
        },
        {
          "cid": "9de5069c5afe602b2ea0a04b66beb2c0",
          "created_timestamp": "2022-11-10T15:15:59Z",
          "id": "9de5069c5afe602b2ea0a04b66beb2c0_9de5069c5afe602b2ea0a04b66beb2c0",
          "origin": "uiproxy",
          "sandbox": [
            {
              "environment_id": 100,
              "sha256": "275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f",
              "submit_name": "https://example.com"
            }
          ],
          "state": "error",
          "user_id": "9de5069c5afe602b2ea0a04b66beb2c0",
          "user_name": "https://example.com",
          "user_uuid": "a149764d-21ee-42af-ac9f-023f5d23ab81"
        }
      ]
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/crowdstrike_falcon_intelligence:1.0.0 run < tests/checkAnalysisStatus.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/CrowdStrike Falcon https://example.com Step name: downloadArtifact\nDownloading an artifact for ID: 30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050...\nGetting authentication token...\n",
    "meta": {},
    "output": {
      "artifacts": [
        {
          "ioc": "44d88612fea8a8f36de82e1278abb02f",
          "source": "extracted",
          "type": "md5"
        },
        {
          "ioc": "44d88612fea8a8f36de82e1278abb02f",
          "source": "extracted",
          "type": "md5"
        },
        {
          "ioc": "44d88612fea8a8f36de82e1278abb02f",
          "source": "extracted",
          "type": "md5"
        },
        {
          "ioc": "44d88612fea8a8f36de82e1278abb02f",
          "source": "extracted",
          "type": "md5"
        },
        {
          "ioc": "3395856ce81f2b7382dee72602f798b642f14140",
          "source": "extracted",
          "type": "sha1"
        },
        {
          "ioc": "3395856ce81f2b7382dee72602f798b642f14140",
          "source": "extracted",
          "type": "sha1"
        },
        {
          "ioc": "3395856ce81f2b7382dee72602f798b642f14140",
          "source": "extracted",
          "type": "sha1"
        },
        {
          "ioc": "3395856ce81f2b7382dee72602f798b642f14140",
          "source": "extracted",
          "type": "sha1"
        },
        {
          "ioc": "275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f",
          "source": "extracted",
          "type": "sha256"
        },
        {
          "ioc": "275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f",
          "source": "extracted",
          "type": "sha256"
        },
        {
          "ioc": "275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f",
          "source": "extracted",
          "type": "sha256"
        },
        {
          "ioc": "275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f",
          "source": "extracted",
          "type": "sha256"
        },
        {
          "ioc": "275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f",
          "source": "input",
          "type": "sha256"
        },
        {
          "ioc": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
          "source": "runtime",
          "type": "sha256"
        },
        {
          "ioc": "275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f",
          "source": "runtime",
          "type": "sha256"
        }
      ]
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/crowdstrike_falcon_intelligence:1.0.0 run < tests/downloadArtifact.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/CrowdStrike Falcon https://example.com Step name: getFullReport\nGetting a list of report summaries for IDs: ['9de5069c5afe602b2ea0a04b66beb2c0_9de5069c5afe602b2ea0a04b66beb2c0', '9de5069c5afe602b2ea0a04b66beb2c0_9de5069c5afe602b2ea0a04b66beb2c0', '9de5069c5afe602b2ea0a04b66beb2c0_9de5069c5afe602b2ea0a04b66beb2c0', '9de5069c5afe602b2ea0a04b66beb2c0_9de5069c5afe602b2ea0a04b66beb2c0']...\nGetting authentication token...\n",
    "meta": {},
    "output": {
      "reports": [
        {
          "cid": "9382986b58cb4bb4935e7eba079842f3",
          "createdTimestamp": "2022-08-30T09:17:31Z",
          "id": "9382986b58cb4bb4935e7eba079842f3_d1ce821d74484545897f2fd5db40d0a7",
          "origin": "uiproxy",
          "sandbox": [
            {
              "environmentDescription": "Windows 7 32 bit",
              "environmentId": 100,
              "errorMessage": "File \"https://example.com\" was detected as \"image\", this format is not supported on WINDOWS",
              "errorOrigin": "CLIENT",
              "errorType": "FILE_TYPE_BAD_ERROR",
              "fileType": "PNG image data, 1682 x 1068, 8-bit/color RGBA, non-interlaced",
              "sha256": "275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f",
              "submissionType": "file",
              "submitName": "https://example.com"
            }
          ],
          "userId": "9de5069c5afe602b2ea0a04b66beb2c0",
          "userName": "https://example.com",
          "verdict": "no verdict"
        },
        {
          "cid": "9de5069c5afe602b2ea0a04b66beb2c0",
          "createdTimestamp": "2022-08-30T10:08:49Z",
          "id": "9de5069c5afe602b2ea0a04b66beb2c0_9de5069c5afe602b2ea0a04b66beb2c0",
          "iocReportBroadCsvArtifactId": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
          "iocReportBroadJsonArtifactId": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
          "iocReportBroadMaecArtifactId": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
          "iocReportBroadStixArtifactId": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
          "iocReportStrictCsvArtifactId": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
          "iocReportStrictJsonArtifactId": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
          "iocReportStrictMaecArtifactId": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
          "iocReportStrictStixArtifactId": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
          "origin": "uiproxy",
          "sandbox": [
            {
              "environmentDescription": "Windows 7 32 bit",
              "environmentId": 100,
              "fileType": "PDF document, version https://example.com",
              "networkSettings": "default",
              "sampleFlags": [
                "Extracted Files"
              ],
              "sha256": "275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f",
              "submissionType": "file",
              "submitName": "https://example.com",
              "verdict": "no specific threat"
            }
          ],
          "userId": "9de5069c5afe602b2ea0a04b66beb2c0",
          "userName": "https://example.com",
          "verdict": "no specific threat"
        },
        {
          "cid": "9de5069c5afe602b2ea0a04b66beb2c0",
          "createdTimestamp": "2022-09-28T07:45:34Z",
          "id": "9de5069c5afe602b2ea0a04b66beb2c0_9de5069c5afe602b2ea0a04b66beb2c0",
          "origin": "uiproxy",
          "sandbox": [
            {
              "environmentDescription": "Windows 7 32 bit",
              "environmentId": 100,
              "errorMessage": "The file \"Dockerfile\" has the file format \"text\", which is not supported",
              "errorOrigin": "CLIENT",
              "errorType": "FILE_TYPE_BAD_ERROR",
              "fileType": "ASCII text",
              "sha256": "275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f",
              "submissionType": "file",
              "submitName": "Dockerfile"
            }
          ],
          "userId": "9de5069c5afe602b2ea0a04b66beb2c0",
          "userName": "https://example.com",
          "verdict": "no verdict"
        },
        {
          "cid": "9de5069c5afe602b2ea0a04b66beb2c0",
          "createdTimestamp": "2022-11-07T14:14:18Z",
          "id": "9de5069c5afe602b2ea0a04b66beb2c0_9de5069c5afe602b2ea0a04b66beb2c0",
          "iocReportBroadCsvArtifactId": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
          "iocReportBroadJsonArtifactId": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
          "iocReportBroadMaecArtifactId": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
          "iocReportBroadStixArtifactId": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
          "iocReportStrictCsvArtifactId": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
          "iocReportStrictJsonArtifactId": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
          "iocReportStrictMaecArtifactId": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
          "iocReportStrictStixArtifactId": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
          "origin": "apigateway",
          "sandbox": [
            {
              "environmentDescription": "Windows 7 32 bit",
              "environmentId": 100,
              "fileType": "PDF document, version https://example.com",
              "networkSettings": "default",
              "sampleFlags": [
                "Extracted Files"
              ],
              "sha256": "275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f",
              "submissionType": "file",
              "submitName": "my_submit_name",
              "verdict": "no specific threat"
            }
          ],
          "userTags": [
            "example"
          ],
          "verdict": "whitelisted"
        }
      ]
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/crowdstrike_falcon_intelligence:1.0.0 run < tests/getFullReport.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/CrowdStrike Falcon https://example.com Step name: getReportsIDs\nGetting a list of sandbox report IDs...\nGetting authentication token...\n",
    "meta": {},
    "output": {
      "reportIds": [
        "9de5069c5afe602b2ea0a04b612eb2c0_9de5069c5afe602b2bb0a04b66beb2c0",
        "9de5069c5afe602b2ea0a04b612eb2c0_9de5069c5afe602b2bb0a04b66beb2c0",
      ]
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/crowdstrike_falcon_intelligence:1.0.0 run < tests/getReportIDs.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/CrowdStrike Falcon https://example.com Step name: getShortReport\nGetting a list of short report summaries for IDs: ['9de5069c5afe602b2ea0a04b66beb2c0_9de5069c5afe602b2ea0a04b66beb2c0', '9de5069c5afe602b2ea0a04b66beb2c0_9de5069c5afe602b2ea0a04b66beb2c0']...\nGetting authentication token...\n",
    "meta": {},
    "output": {
      "reports": [
        {
          "cid": "9de5069c5afe602b2ea0a04b66beb2c0",
          "createdTimestamp": "2022-09-28T07:45:34Z",
          "id": "9de5069c5afe602b2ea0a04b66beb2c0_9de5069c5afe602b2ea0a04b66beb2c0",
          "origin": "uiproxy",
          "sandbox": [
            {
              "environmentDescription": "Windows 7 32 bit",
              "environmentId": 100,
              "errorMessage": "The file \"Dockerfile\" has the file format \"text\", which is not supported",
              "errorOrigin": "CLIENT",
              "errorType": "FILE_TYPE_BAD_ERROR",
              "fileType": "ASCII text",
              "sha256": "275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f",
              "submissionType": "file",
              "submitName": "Dockerfile"
            }
          ],
          "userId": "9de5069c5afe602b2ea0a04b66beb2c0",
          "userName": "https://example.com",
          "verdict": "no verdict"
        },
        {
          "cid": "9de5069c5afe602b2ea0a04b66beb2c0",
          "createdTimestamp": "2022-11-07T12:59:53Z",
          "id": "9de5069c5afe602b2ea0a04b66beb2c0_9de5069c5afe602b2ea0a04b66beb2c0",
          "iocReportBroadCsvArtifactId": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
          "iocReportBroadJsonArtifactId": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
          "iocReportBroadMaecArtifactId": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
          "iocReportBroadStixArtifactId": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
          "iocReportStrictCsvArtifactId": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
          "iocReportStrictJsonArtifactId": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
          "iocReportStrictMaecArtifactId": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
          "iocReportStrictStixArtifactId": "30f800f97aeaa8d62bdf3a6fb2b0681179a360c12e127f07038f8521461e5050",
          "origin": "apigateway",
          "sandbox": [
            {
              "environmentDescription": "Windows 7 32 bit",
              "environmentId": 100,
              "fileType": "PDF document, version https://example.com",
              "networkSettings": "default",
              "sampleFlags": [
                "Extracted Files"
              ],
              "sha256": "275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f",
              "submissionType": "file_url",
              "submitName": "https://example.com",
              "submitUrl": "hxxps://https://example.com",
              "verdict": "no specific threat"
            }
          ],
          "userTags": [
            "example"
          ],
          "verdict": "whitelisted"
        }
      ]
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/crowdstrike_falcon_intelligence:1.0.0 run < tests/getShortReport.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/CrowdStrike Falcon https://example.com Step name: getSubmissionsIDs\nGetting a list of submission IDs...\nGetting authentication token...\n",
    "meta": {},
    "output": {
      "submissionIds": [
        "9382986b58cb4b44935c2eba079842f3_0a54f7a33701461299899f294c2eb53d",
        "9382986b58cb4b44935c2eba079842f3_a17b8037186f4c8a9129d88b9d40fc9d",
        "9382986b58cb4b44935c2eba079842f3_11f08e0a490d4bc9912a7375af4664bc",
      ]
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/crowdstrike_falcon_intelligence:1.0.0 run < tests/getSubmissionsIDs.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nrapid7/CrowdStrike Falcon https://example.com Step name: submitAnalysis\nSubmitting ana1ysis...\nGetting authentication token...\n",
    "meta": {},
    "output": {
      "submission": {
        "cid": "9de5069c5afe602b2ea0a04b66beb2c0",
        "createdTimestamp": "2022-11-22T06:10:42Z",
        "id": "9de5069c5afe602b2ea0a04b66beb2c0_9de5069c5afe602b2ea0a04b66beb2c0",
        "origin": "apigateway",
        "sandbox": [
          {
            "actionScript": "default_randomfiles",
            "environmentId": 100,
            "networkSettings": "default",
            "sha256": "275a021bbfb6489e54d471899f7db9d1663fc695ec2fe2a2c4538aabf651fd0f",
            "submitName": "plugin_action_submission",
            "systemDate": "2022-09-22",
            "systemTime": "12:55"
          }
        ],
        "state": "created"
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/crowdstrike_falcon_intelligence:1.0.0 run < tests/submitAnalysis.json
</summary>
</details>

### UI

Screenshots of the plugin being used in Komand for validating use go below this line.

### CODE REVIEW

**Review by dsliwinski-r7 — APPROVED** [review]

File: plugins/crowdstrike_falcon_intelligence/icon_crowdstrike_falcon_intelligence/actions/getSubmissionsIDs/action.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        super(self.__class__, self).__init__(
+            name="getSubmissionsIDs",
+            description=Component.DESCRIPTION,
+            input=GetSubmissionsIDsInput(),
+            output=GetSubmissionsIDsOutput(),
+        )
+
+    def run(self, params: dict = None):
+        limit = params.get(Input.LIMIT, 5000)
+        if limit > 5000:
```
Comment (jkaluza-rapid7) [inline plugins/crowdstrike_falcon_intelligence/icon_crowdstrike_falcon_intelligence/actions/getSubmissionsIDs/action.py:18]: You could move `5000` to a constant

File: plugins/crowdstrike_falcon_intelligence/icon_crowdstrike_falcon_intelligence/actions/getSubmissionsIDs/action.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            self.logger.warning(
+                "Provided limit value is larger than 5000 but this action will return up to 5000 results."
+            )
```
Comment (jkaluza-rapid7) [inline plugins/crowdstrike_falcon_intelligence/icon_crowdstrike_falcon_intelligence/actions/getSubmissionsIDs/action.py:21]: I think that way of handling the limit could potentially confuse the customer. Maybe we should throw an error, or use some paging if its available in their API?

**Review by jkaluza-rapid7 — APPROVED** [review]

File: plugins/crowdstrike_falcon_intelligence/icon_crowdstrike_falcon_intelligence/actions/getReportsIDs/action.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+            input=GetReportsIDsInput(),
+            output=GetReportsIDsOutput(),
+        )
+
+    def run(self, params: dict = None):
+        limit = params.get(Input.LIMIT, 5000)
+        if limit > 5000:
+            self.logger.warning(
+                "Provided limit value is larger than 5000 but this action will return up to 5000 results."
+            )
```
Comment (igorski-r7) [inline plugins/crowdstrike_falcon_intelligence/icon_crowdstrike_falcon_intelligence/actions/getReportsIDs/action.py:21]: Please move that `5000` to const variable

File: plugins/crowdstrike_falcon_intelligence/icon_crowdstrike_falcon_intelligence/actions/getSubmissionsIDs/action.py (hunk 2 above)
Comment (igorski-r7) [inline plugins/crowdstrike_falcon_intelligence/icon_crowdstrike_falcon_intelligence/actions/getSubmissionsIDs/action.py:21]: Same here. Please move that `5000` to the const variable 😃

File: plugins/crowdstrike_falcon_intelligence/icon_crowdstrike_falcon_intelligence/util/api.py [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+            raise PluginException(preset=PluginException.Preset.UNKNOWN, data=e)
+
+    def make_json_request(
+        self, method: str, url: str, params: dict = None, json_data: dict = None
+    ) -> Union[list, dict]:
+        try:
+            response = self.make_request(method=method, url=url, params=params, json_data=json_data)
+            return response.json()
+        except json.decoder.JSONDecodeError as e:
+            raise PluginException(preset=PluginException.Preset.INVALID_JSON, data=e)
```
Comment (igorski-r7) [inline plugins/crowdstrike_falcon_intelligence/icon_crowdstrike_falcon_intelligence/util/api.py:164]: Just to stay with one concept. Kindly please do not use single variable names 😄 
```suggestion
        except json.decoder.JSONDecodeError as error:
            raise PluginException(preset=PluginException.Preset.INVALID_JSON, data=error)
```

**Review by igorski-r7 — CHANGES_REQUESTED** [review]
In general it looks great! 😃 I think some minor changes are needed. Left small comments.

**Review by igorski-r7 — APPROVED** [review]

**Comment by dlaverty-r7** [comment]
@mgrandys-r7 , do you mind checking the pluginValidator failure (it may just be the known timestamp issue) and resolve the conflict in the plugins/azure_ad_admin/icon_azure_ad_admin/actions/change_user_password/action.py and we can get this one merged?

**Comment by mgrandys-r7** [comment]
> @mgrandys-r7 , do you mind checking the pluginValidator failure (it may just be the known timestamp issue) and resolve the conflict in the plugins/azure_ad_admin/icon_azure_ad_admin/actions/change_user_password/action.py and we can get this one merged?

@dlaverty-r7 I've just rebased this branch so it's ready to be merged. As you mentioned the PluginValidator fails because of (in)correct date format.

**Pull request merged**